sci-wcalc/air_coil_syn.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH air_coil_syn "" "" "Wcalc" "Wcalc Commands"
.B air_coil_syn
- Single layer air core solenoid inductor synthesis

.SH SYNOPSIS

[N_out, len_out, fill_out] = 
air_coil_syn(L, len, fill, AWG, rho, dia, freq, flag)


.SH INPUT PARAMETERS

.TP 10
L
: Desired inductance (Henries)
.TP 10
len
: length of the coil (meters)
.TP 10
fill
: ratio of coil length to closewound length
.TP 10
AWG
: wire size (American Wire Gauge)
.TP 10
rho
: resistivity of wire (ohm-m)
.TP 10
dia
: inside diameter of coil (meters)
.TP 10
freq
: analysis frequency (Hertz)
.TP 10
flag
: indicates which parameters should be optimized
    
.RS
.TP 10
flag=0
: the minimum number of turns is calculated
	along with the corresponding required length/fill.  The diameter is
	held constant.
.TP 10
flag=1
: the number of turns is fixed at the value
        given and the required length/fill is calculated.  The
	diameter is held constant.
.RE


.SH OUTPUT VALUES

.TP 10
N_out
: calculated required number of turns
.TP 10
len_out
: calculated required length (meters)
.TP 10
fill_out
: calculated required fill
.SH DESCRIPTION

Function to synthesize the physical parameters of a single layer,
uniformly spaced, air core, solenoid inductor.  The physical
dimensions of the coil are synthesized from an inductance 
specification.

.SH EXAMPLE
.nf

.TP
 // we want 220 nH
L=220e-9;
.TP
 // we'll synthesize the number of turns
N=0;
.TP
 // we'll synthesize the length/fill
len=0;
.TP
 // we'll synthesize the length/fill
fill=0;
.TP
 // wire size
AWG=22;
.TP
 // Resistivity of copper (ohm-m)
rho=1.72e-8;
.TP
 // 0.25 inches inside diameter
dia=0.25*0.0254;
.TP
 // 10 MHz operation
f=10e6;
.TP
 // synthesize for minimum number of turns
flag=0;
[Nout,len_out,fill_out] = air_coil_syn(L,N,len,fill,AWG,rho,dia,f,flag);
disp(sprintf('Number of turns = %g ',Nout));
disp(sprintf('Length=%g inches',len_out/0.0254));
disp(sprintf('Fill=%g',fill_out));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/air_coil_syn.m
%AIR_COIL_SYN     - Single layer air core solenoid inductor synthesis 
%  
%  SYNOPSIS
%  
%  [N_out, len_out, fill_out] = air_coil_syn(L, len, fill, AWG, rho, dia, freq,
%  flag)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * L : Desired inductance (Henries)
%    * len : length of the coil (meters)
%    * fill : ratio of coil length to closewound length
%    * AWG : wire size (American Wire Gauge)
%    * rho : resistivity of wire (ohm-m)
%    * dia : inside diameter of coil (meters)
%    * freq : analysis frequency (Hertz)
%    * flag : indicates which parameters should be optimized
%        + flag=0 : the minimum number of turns is calculated along with the
%          corresponding required length/fill. The diameter is held constant.
%        + flag=1 : the number of turns is fixed at the value given and the
%          required length/fill is calculated. The diameter is held constant.
%  
%  OUTPUT VALUES
%  
%    * N_out : calculated required number of turns
%    * len_out : calculated required length (meters)
%    * fill_out : calculated required fill
%  
%  DESCRIPTION
%  
%  Function to synthesize the physical parameters of a single layer, uniformly
%  spaced, air core, solenoid inductor. The physical dimensions of the coil are
%  synthesized from an inductance specification.
%  
%  EXAMPLE
%  
%   % we want 220 nH
%  L=220e-9;
%   % we'll synthesize the number of turns
%  N=0;
%   % we'll synthesize the length/fill
%  len=0;
%   % we'll synthesize the length/fill
%  fill=0;
%   % wire size
%  AWG=22;
%   % Resistivity of copper (ohm-m)
%  rho=1.72e-8;
%   % 0.25 inches inside diameter
%  dia=0.25*0.0254;
%   % 10 MHz operation
%  f=10e6;
%   % synthesize for minimum number of turns
%  flag=0;
%  [Nout,len_out,fill_out] = air_coil_syn(L,N,len,fill,AWG,rho,dia,f,flag);
%  disp(sprintf('Number of turns = %g ',Nout));
%  disp(sprintf('Length=%g inches',len_out/0.0254));
%  disp(sprintf('Fill=%g',fill_out));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
